utils/polus-imagej-util/generate.py

```diff
@@ -0,0 +1,138 @@
+    cookiecutter_path = base_path.joinpath('cookiecutter.json')
+
+    # Get list of all plugin directories in the cookietin directory 
+    plugins = list(cookietin_path.iterdir()) 
+    
+    # Create path to imagej testing directory
+    test_dir = Path(cwd.joinpath('imagej-testing'))
+    
+    # Check if the imagej testing directory already exists
+    if test_dir.exists():
+        # Remove directory if it exists
+        shutil.rmtree(test_dir)
+    
+    # Create the imagej testing directory
+    os.mkdir(test_dir)
+    
+    # Create path to the imagej shell script testing file
+    shell_test_path = test_dir.joinpath('shell_test.py')
+    
+    # Create test summary file
+    test_summary_path = shell_test_path.with_name('test-summary.log')
+
+    # Create testing shell script file
+    with open(shell_test_path, 'w') as fhand:
+        fhand.write('import os, sys\nfrom pathlib import Path\nsrc_path = Path(__file__).parents[1]\nsys.path.append(str(src_path))\n')
+        fhand.close()
+    
+    # Creater counters for plugins and ops
+    pluging_count = 0
+    op_count = 0
+    
+    # Iterate over all plugin directories in the cookietin directory 
+    for plugin in plugins:
+        
+        #plugins_to_generate = ['math-add']
+        #plugins_to_generate = ['image-integral', 'image-distancetransform', 'filter-dog']
+        #plugins_to_generate += ['filter-dog', 'filter-addNoise', 'filter-convolve', 'filter-bilateral', 'filter-correlate']
+        #if plugin.name in plugins_to_generate:
+        if True:
+             
+            # Create a path for the plugin
+            path = Path(os.getcwd()).joinpath('polus-imagej-' + plugin.name.lower() + '-plugin')
+            
+            # If the plugin path is already a directory remove it and its children
+            if path.exists():
+                shutil.rmtree(path)
+            
+            # Move the cookiecutter.json file for the current plug in to the polus-imagej-util directory overwriting last plugin json file
+            shutil.copy(str(plugin.joinpath('cookiecutter.json')), cookiecutter_path)
+            
+            # Run the cookiecutter utility for the plugin
+            os.system('cookiecutter ./utils/polus-imagej-util/ --no-input')
+            
+            # Open the shell script in append mode
+            with open(shell_test_path, 'a') as fhand:
+                
+                # Get plugin dictionary key
+                plugin_key = plugin.name.replace('-', '.')
+                
+                # Get all available ops for the plugin
+                ops = [op for op in populater._namespaces[plugin_key].supportedOps.keys()]
+                
+                # Create a list of the operating sytem commands
+                #commands = ["python "+str(path)+"/tests/unit_test.py --opName "+op for op in ops]
+                commands = ["python "+str(path)+"/tests/unit_test.py --opName '{}'".format(op) for op in ops]
```

Comment:
This will work, but is it possible to actually use the `unittest` package to configure the running of all unit tests?